hazelcast-hibernate53/src/main/java/com/hazelcast/hibernate/distributed/IMapRegionCache.java

```diff
@@ -77,9 +80,12 @@ public Object get(final Object key, final long txTimestamp) {
         return entry == null ? null : entry.getValue(txTimestamp);
     }
 
+    /**
+     * Returns the number of in-memory entries (without backups) or -1 if using Hazelcast Client
+     */
     @Override
     public long getElementCountInMemory() {
-        return map.size();
+        return isMember ? map.getLocalMapStats().getOwnedEntryCount() : -1;
```

Comment:
I wonder your thoughts if this is necessary to be mentioned (member based, not cluster wide entry count) on the documentation.

Reply:
It definitely should. Although `-1` should be pretty intuitive to interpret for someone that sees it for the first time